tests/e2e/models/test_andromede_models.py

```diff
@@ -0,0 +1,153 @@
+from andromede.study.resolve_components import (
+    build_data_base,
+    build_network,
+    resolve_system,
+)
+
+
+@pytest.mark.parametrize(
+    "system_file, optim_result_file, timespan, batch, relative_accuracy, results_dir, systems_dir, series_dir",
+    [
+        (
+            "dsr_validation.yml",
+            "dsr_case_results.csv",
+            168,
+            20,
+            1e-6,
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+        (
+            "base_validation.yml",
+            "base_case_results.csv",
+            168,
+            20,
+            1e-6,
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+        (
+            "electrolyser_validation.yml",
+            "electrolyser_case_results.csv",
+            168,
+            20,
+            1e-6,
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+        (
+            "storage_validation.yml",
+            "storage_case_results.csv",
+            168,
+            20,
+            1e-6,
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+        (
+            "bde_system.yml",
+            "bde_case_results.csv",
+            168,
+            20,
+            1e-6,
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+        (
+            "cluster_validation_1.yml",
+            "cluster_testing1_results.csv",
+            168,
+            20,
+            1e-6,
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+        (
+            "cluster_validation_2.yml",
+            "cluster_testing2_results.csv",
+            168,
+            20,
+            1e-4,  # Default Fico XPRESS Tolerance
+            "tests/e2e/models/results/",
+            "tests/e2e/models/systems/",
+            "tests/e2e/models/series/",
+        ),
+    ],
+)
+def test_model_behaviour(
+    system_file: str,
+    optim_result_file: str,
+    timespan: int,
+    batch: int,
+    relative_accuracy: float,
+    results_dir: Path,
+    systems_dir: Path,
+    series_dir: Path,
+) -> None:
+    scenarios = 1
+
+    with open(systems_dir + system_file) as compo_file:
+        with open(
+            "src/andromede/libs/antares_historic/antares_historic.yml"
+        ) as lib_file1:
+            with open(
+                "src/andromede/libs/reference_models/andromede_v1_models.yml"
+            ) as lib_file2:
+                input_libraries = [
+                    parse_yaml_library(lib_file1),
+                    parse_yaml_library(lib_file2),
+                ]
```

Comment:
```suggestion
        with open(
            "src/andromede/libs/antares_historic/antares_historic.yml"
        ) as lib_file1:
            lib_historic = parse_yaml_library(lib_file1)
        with open(
            "src/andromede/libs/reference_models/andromede_v1_models.yml"
        ) as lib_file2:
            lib_v1 = parse_yaml_library(lib_file2)
        input_libraries = [
            parse_yaml_library(lib_file1),
            parse_yaml_library(lib_file2),
            ]
```
You should put under the `with` only the code that needs to have the files open, as when we exit the `with` part the file is closed. In the current state, the two files stay open for the run

Reply:
Done